### PR TITLE
decompress method.

### PR DESCRIPTION
**Pull request by kxk302**
Hi @wm75 
I wrote a method to decompress the covid file. Could you please review it when you have a chance?
Thank you!

### CODE REVIEW

File: decompress.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import argparse
+import itertools
```
Comment (wm75) [inline decompress.py:2]: unused import

File: decompress.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import json
+
+from compress_for_dashboard import parse_compressed
+
+if __name__ == "__main__":
+  ap = argparse.ArgumentParser()
+  ap.add_argument("-i", "--input", help="Name of the file to decompress", required=True)
+  ap.add_argument("-o", "--output", help="Name of the decompressed file", required=True)
+  args =vars( ap.parse_args())
```
Comment (wm75) [inline decompress.py:11]: not needed if you stick to dot-qualifiers

File: decompress.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+  with open(args['input'], "r") as in_f, open(args['output'], "w") as out_f:
```
Comment (wm75) [inline decompress.py:13]: args.input and args.output (see above)

File: decompress.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    compressed_data = json.load(in_f)
+  
+    # A row has the following columns
+    # "Sample","POS","REF","ALT","AF","EFFECT","CODON","TRID","AA"
+    header = "Sample\tPOS\tREF\tALT\tAF\tEFFECT\tCODON\tTRID\tAA\n"
+    out_f.write(header)
+
+    rows = list(parse_compressed(compressed_data))
+    for row in rows:
```
Comment (wm75) [inline decompress.py:22]: `for row in parse_compressed(compressed_data):`

no need to turn things into a list

File: decompress.py [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+      sample = row[0]
+      pos = row[1][0]
+      ref = row[1][1]
+      alt = row[1][2]
+      af = row[1][3]
+      effect = row[1][4]
+      codon = row[1][5]
+      trid = row[1][6]
```
Comment (wm75) [inline decompress.py:30]: `pos, ref, alt, af, effect, codon, trid = row[1]`

File: decompress.py [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+      aa = row[2]
+      out_f.write(str(sample) + "\t" + str(pos) + "\t" + str(ref) + "\t" + str(alt) + "\t" + str(af) + "\t" + str(effect) + "\t" + str(codon) + "\t" + str(trid) + "\t" + str(aa) + "\n")
```
Comment (wm75) [inline decompress.py:32]: out_f.write(
    '\t'.join([sample, str(pos), ref, alt, str(af), effect, codon, trid, aa]) + '\n')
)

File: decompress.py [hunk 7]
```diff
@@ -0,0 +1,32 @@
+import argparse
+import itertools
+import json
+
+from compress_for_dashboard import parse_compressed
+
+if __name__ == "__main__":
+  ap = argparse.ArgumentParser()
+  ap.add_argument("-i", "--input", help="Name of the file to decompress", required=True)
+  ap.add_argument("-o", "--output", help="Name of the decompressed file", required=True)
```
Comment (wm75) [inline decompress.py:10]: maybe just use positional args here for consistency with the compress_for_dashboard CLI?

**Review by wm75 — CHANGES_REQUESTED** [review]
You will need to target the `aggregate-observable-data` branch to have the import work.

File: decompress.py (hunk 1 above)
Comment (kxk302) [reply decompress.py:2]: Removed it. Thx.

File: decompress.py (hunk 2 above)
Comment (kxk302) [reply decompress.py:11]: Removed vars() call

File: decompress.py (hunk 3 above)
Comment (kxk302) [reply decompress.py:13]: Revised accordingly.

File: decompress.py (hunk 4 above)
Comment (kxk302) [reply decompress.py:22]: Removed list(). Renamed rows to rows_gen

File: decompress.py (hunk 5 above)
Comment (kxk302) [reply decompress.py:30]: Good idea. More compact. Revised accordingly.

File: decompress.py (hunk 6 above)
Comment (kxk302) [reply decompress.py:32]: Revised accordingly.

File: decompress.py (hunk 7 above)
Comment (kxk302) [reply decompress.py:10]: Sure.

**Comment by kxk302** [comment]
> You will need to target the `aggregate-observable-data` branch to have the import work.

@wm75 I have updated the file based on your feedback (Thank You!). Could you please review it and if all is good, I'll close this PR and create a new one against the correct branch?

**Comment by kxk302** [comment]
> > You will need to target the `aggregate-observable-data` branch to have the import work.
> 
> @wm75 I have updated the file based on your feedback (Thank You!). Could you please review it and if all is good, I'll close this PR and create a new one against the correct branch?

Here is the new PR: https://github.com/usegalaxy-eu/ena-cog-uk-wfs/pull/24

**Comment by kxk302** [comment]
Ill close this PR, to avoid confusion. Work is being reviewed in https://github.com/usegalaxy-eu/ena-cog-uk-wfs/pull/24

**Pull request closed without merging**